### PR TITLE
Allow PE Yocto users to modify the mbed-cloud-client config options

### PR DESCRIPTION
**Pull request by ygoyal18**
Customer has requested a way to modify the keep alive interval through Yocto recipe. Reduced the default lifetime (registration message interval) from 1 hour to 30 mins.

### CODE REVIEW

**Review by x85446 — APPROVED** [review]
So to overide they will provide a bbappend file with a new recipes-connectivity/mbed-edge-core/files/mbed_cloud_client_user_config.h file in their meta.

**Comment by cmonr** [comment]
Does this need to come into the 2.2 release?

**Comment by ygoyal18** [comment]
We can hold on to this for later release, if it seems risky. If a customer wants to try, we can give this patch for reference.

**Comment by ygoyal18** [comment]
@esajaa @JanneKiiskila Please review this as well.

**Review by esajaa — APPROVED** [review]

File: recipes-connectivity/mbed-edge-core/files/mbed_cloud_client_user_config.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/*
+ * ----------------------------------------------------------------------------
+ * Copyright 2018 ARM Ltd.
```
Comment (JanneKiiskila) [inline recipes-connectivity/mbed-edge-core/files/mbed_cloud_client_user_config.h:3]: 2018-2021

**Review by JanneKiiskila — APPROVED** [review]

**Pull request merged**